reactioncommerce/reaction
# Fix 3701 nnnnat email status

Resolves #3701 & #3685   
Impact: major  
Type: bugfix

## Issue
Mail provided status was not updating in real-time if the status had changed between `valid` or `error`. The [EmailConfig](https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/core/email/client/containers/emailConfig.js) container was only checking the email settings status in the `componentWillMount` method and was unaware if the status changed after the component was mounted.

## Solution
I moved the logic to check the email setting status into it's own method and call it from both `componentWillMount` and `componentWillUpdate` methods. 

## Testing
1. As an Admin open the email admin panel to view the current Mail Provider status.
2. Select the edit button and update the email settings to have a incorrect username or password. 
3. Click save and the Mail Provider status should now display as red/error.
3. Edit the email settings again by entering the correct username or password.
4. Click save and the Mail Provider status should now display as green/valid.
